File: leetcode/python/48. Rotate Image.py
def rotate(arr):
  n = len(arr)
  for i in range(n//2):
    for j in range(-(-n//2)):
      temp = arr[i][j]
      arr[i][j] = arr[n-1-j][i]
      arr[n - 1 - j][i] = arr[n - 1 - i][n - 1 - j]
      arr[n - 1 - i][n - 1 - j] = arr[j][n - 1 - i]
      arr[j][n - 1 - i] = temp

arr = [
  [1, 2, 3],
  [4, 5, 6],
  [7, 8, 9],  
]

print(arr)
rotate(arr)
print("#############")
print(arr)

File: leetcode/kotlin/48. Rotate Image.kt
import java.io.*

object Main {
    var ps: PrintStream = System.out
    fun rotate(matrix: Array<IntArray>) {
        val n = matrix.size
        for (i in 0 until n / 2) {
            for (j in 0 until Math.ceil(n.toDouble() / 2.0)) {
                val temp = matrix[i][j]
                matrix[i][j] = matrix[n - 1 - j][i]
                matrix[n - 1 - j][i] = matrix[n - 1 - i][n - 1 - j]
                matrix[n - 1 - i][n - 1 - j] = matrix[j][n - 1 - i]
                matrix[j][n - 1 - i] = temp
            }
        }
    }

    fun display(arr: Array<IntArray>) {
        for (i in arr) {
            for (j in i) {
                ps.print("$j ")
            }
            ps.println("")
        }
    }

    fun main(vararg s: String?) {
        val arr = arrayOf(intArrayOf(1, 2, 3), intArrayOf(4, 5, 6), intArrayOf(7, 8, 9))
        display(arr)
        rotate(arr)
        ps.println("################")
        display(arr)
    }
}